### PR TITLE
Initial Nebula Stats Exporter

### PR DESCRIPTION
**Pull request by knightXun**
this is initial Nebula Stats Exporter Pr. this PR convert nebula original metrics to promethues metrics.

### CODE REVIEW

**Comment by knightXun** [comment]
ping @yixinglu

**Comment by knightXun** [comment]
ping @yixinglu, this can be done

File: deploy/bare-metal/nebula.json [hunk 1]
```diff
@@ -0,0 +1,1078 @@
+{
```
Comment (yixinglu) [inline deploy/bare-metal/nebula.json]: We can add a `grafana` folder and move this json file into it for easy understanding.

File: deploy/kubernetes/nebula_exporter.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      annotations:
+        prometheus.io/port: "9100"
+        prometheus.io/scrape: "true"
+      labels:
+        k8s-app: nebula-stats-exporter
+    spec:
+      serviceAccountName: nebula-stats-exporter
+      containers:
+        - name: exporter
+          image: knightxun/nebula-stats-exporter:v0.0.1
```
Comment (yixinglu) [inline deploy/kubernetes/nebula_exporter.yaml]: unify docker image name to `vesoft/nebula-stats-exporter:v0.0.1`

File: exporter/exporter.go [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+					}
+
+					if metric, ok := exporter.metadMap[seps[0]]; ok {
+						ch <- prometheus.MustNewConstMetric(metric.desc, prometheus.GaugeValue, values,
+							pod.Name,
+							pod.Namespace,
+							"metad",
+						)
+					}
+				}
```
Comment (yixinglu) [inline exporter/exporter.go]: Why not to extract some common functions for collecting `graph`/`meta`/`storage` metrics from static and k8s server ?

File: go.sum [hunk 4]
```diff
@@ -0,0 +1,230 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (yixinglu) [inline go.sum]: I think this sum file is too messy, and it don't just include your needed dependencies. you can delete it and rebuild the project to generate it again.

File: exporter/exporter.go [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+
+type metrics struct {
+	MetricsType 	string
+	desc 			*prometheus.Desc
+}
+
+type NebulaExporter struct {
+	sync.Mutex
+	Client    		*kubernetes.Clientset
+	mux             *http.ServeMux
```
[the remark this answers is not in the document]
Comment (knightXun) [reply exporter/exporter.go]: this is done in #2

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,5 @@
+FROM centos:7
```
[the remark this answers is not in the document]
Comment (knightXun) [reply Dockerfile]: this is done in #2, change centos:7 to busybox

File: deploy/bare-metal/README.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    endpointIP: 192.168.0.102
+    endpointPort: 32787
+    nebulaType: metad
+```
+
+### 运行nebula-stats-exporter 
+直接运行：
+```bash
+docker run -p --restart=always -p 9100:9100 -v {path to config.yaml}:/config \
+ nebula-stats-exporter:v0.0.1  --bare-metal --bare-metal-config-path=/config/config.yaml
```
[the remark this answers is not in the document]
Comment (knightXun) [reply deploy/bare-metal/README.md]: this is done in #2, change centos:7 to busybox

File: deploy/bare-metal/nebula.json (hunk 1 above)
Comment (knightXun) [reply deploy/bare-metal/nebula.json]: this is done in #2, change centos:7 to busybox

File: deploy/kubernetes/nebula_exporter.yaml (hunk 2 above)
Comment (knightXun) [reply deploy/kubernetes/nebula_exporter.yaml]: this is done in #2,

File: exporter/exporter.go (hunk 3 above)
Comment (knightXun) [reply exporter/exporter.go]: this is done in #2,  extrace a function to collect metrics

File: go.sum (hunk 4 above)
Comment (knightXun) [reply go.sum]: this is done in #2, removed go.sum

**Pull request closed without merging**